extensions/kafka/src/test/java/com/hazelcast/jet/kafka/impl/StreamKafkaPTest.java

```diff
@@ -157,6 +167,159 @@ public void when_projectionFunctionProvided_thenAppliedToReadRecords() {
         }, 5);
     }
 
+    @Test
+    public void when_partitionsInitialOffsetsProvided_thenRecordsReadFromAppropriatePosition() {
```

Comment:
It's kind of questionable what should happen when there's a new partition detected at runtime, for which there's an offset specified in the TopicsConfig. We can use the offset for the new topic, because the user likely didn't get that offset out of thin air, but knew it. But it also makes sense to seek to beginning, as we do for all partitions not existing at the start of the job - the partition didn't exist when the job started, after all. 

Current behavior is to seek to beginning. I think both approaches have their rationale, so I'll stick with the code we have now. But we need to cover it in a test, see `when_partitionAdded_then_consumedFromBeginning()`.

We need one test to test the case with kafka partitions added at runtime: the new partition should seek to beginning even if initial offset is specified. I think the code works like this currently, but we need to check it in a test.

Reply:
I've added `when_partitionAddedWhilePartitionsInitialOffsetsProvided_then_consumedFromBeginning` test.